### PR TITLE
Fix `std::path::Path::file_name()` doc

### LINKED ISSUE #34632: Wrong documentation for std::path::Path::file_name
The documentation states:

> The final component of the path, if it is a normal file.
> If the path terminates in ., .., or consists solely of a root of prefix, file_name will return None.

but:

```
fn do_file_name(x: &str) {
    println!("file_name({}) = {:?}", x, std::path::Path::new(x).file_name());
}

fn main() {
    use std::path::Path;
    do_file_name("/path/with/dot/.");
    do_file_name("/path/with/with/dot/slash/./");
    do_file_name("/path/with/dot/double/slash/.//");
    do_file_name("/path/with/double/dot/..");
    do_file_name("/path/with/nul/bytes/test\0invalid");
}
```

prints

```
file_name(/path/with/dot/.) = Some("dot")
file_name(/path/with/with/dot/slash/./) = Some("slash")
file_name(/path/with/dot/double/slash/.//) = Some("slash")
file_name(/path/with/double/dot/..) = None
file_name(/path/with/nul/bytes/testinvalid) = Some("test\u{0}invalid")
```

(Playground: https://play.rust-lang.org/?gist=2eddf89b4c80926240b9cbd4d6d40821&version=stable&backtrace=0)

I think the documentation for the "ends with ." case is wrong.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #34632

r? @steveklabnik

### CODE REVIEW

File: src/libstd/path.rs [hunk 1]
```diff
@@ -1542,6 +1541,11 @@ impl Path {
     /// let os_str = OsStr::new("foo.txt");
     ///
     /// assert_eq!(Some(os_str), path.file_name());
+    ///
+    /// // Other examples:
```
Comment (steveklabnik) [inline src/libstd/path.rs]: can we put these in a separate doc block rather than extending this one?

**Comment by steveklabnik** [comment]
@bors: r-

```

---- path::Path::file_name_1 stdout ----
	<anon>:4:21: 4:31 error: failed to resolve. Use of undeclared type or module `OsStr` [E0433]
<anon>:4     assert_eq!(Some(OsStr::new("foo.txt")), Path::new("foo.txt/.").file_name());
                             ^~~~~~~~~~
<anon>:4:5: 4:81 note: in this expansion of assert_eq! (defined in <std macros>)
<anon>:4:45: 4:54 error: failed to resolve. Use of undeclared type or module `Path` [E0433]
<anon>:4     assert_eq!(Some(OsStr::new("foo.txt")), Path::new("foo.txt/.").file_name());
                                                     ^~~~~~~~~
<anon>:4:5: 4:81 note: in this expansion of assert_eq! (defined in <std macros>)
<anon>:5:17: 5:27 error: failed to resolve. Use of undeclared type or module `OsStr` [E0433]
<anon>:5 assert_eq!(Some(OsStr::new("foo.txt")), Path::new("foo.txt/.//").file_name());
                         ^~~~~~~~~~
<anon>:5:1: 5:79 note: in this expansion of assert_eq! (defined in <std macros>)
<anon>:5:41: 5:50 error: failed to resolve. Use of undeclared type or module `Path` [E0433]
<anon>:5 assert_eq!(Some(OsStr::new("foo.txt")), Path::new("foo.txt/.//").file_name());
                                                 ^~~~~~~~~
<anon>:5:1: 5:79 note: in this expansion of assert_eq! (defined in <std macros>)
<anon>:6:18: 6:27 error: failed to resolve. Use of undeclared type or module `Path` [E0433]
<anon>:6 assert_eq!(None, Path::new("foo.txt/..").file_name());
                          ^~~~~~~~~
<anon>:6:1: 6:55 note: in this expansion of assert_eq! (defined in <std macros>)
error: aborting due to previous error(s)
thread 'path::Path::file_name_1' panicked at 'Box<Any>', src/librustc/session/mod.rs:170
note: Run with `RUST_BACKTRACE=1` for a backtrace.


```

**Comment by GuillaumeGomez** [comment]
@steveklabnik: I put back the imports.

**Pull request merged**